src/components/input/test/Checkbox-test.js

```diff
@@ -0,0 +1,47 @@
+import { mount } from 'enzyme';
+import { act } from 'preact/test-utils';
+
+import { testCompositeComponent } from '../../test/common-tests';
+
+import Checkbox from '../Checkbox';
+
+const createComponent = (Component, props = {}) => {
+  return mount(<Component {...props}>This is child content</Component>);
+};
+
+describe('Checkbox', () => {
+  testCompositeComponent(Checkbox, { elementSelector: 'input' });
+
+  it('shows an icon representing checkbox state', () => {
+    const wrapper = createComponent(Checkbox);
+
+    assert.isTrue(wrapper.find('CheckboxOutlineIcon').exists());
+    assert.isFalse(wrapper.find('CheckboxCheckedIcon').exists());
+
+    wrapper.setProps({ checked: true });
+
+    assert.isFalse(wrapper.find('CheckboxOutlineIcon').exists());
+    assert.isTrue(wrapper.find('CheckboxCheckedIcon').exists());
+  });
+
+  it('calls provided `onChange` callback when input changes', () => {
+    const onChange = sinon.stub();
+    const wrapper = createComponent(Checkbox, {
+      onChange,
+    });
+
+    const checkbox = wrapper.find('input');
+    assert.isNotOk(checkbox.getDOMNode().getAttribute('checked'));
+
+    act(() => {
+      // See https://github.com/preactjs/enzyme-adapter-preact-pure/issues/123#issuecomment-699474177
+      const el = checkbox.getDOMNode();
+      el.setAttribute('checked', true);
+      el.dispatchEvent(new Event('change'));
+    });
+    wrapper.update();
+
+    assert.calledOnce(onChange);
+    assert.equal(checkbox.getDOMNode().getAttribute('checked'), 'true');
```

Comment:
HALLLLPPPPPP

Reply:
Maybe something like this:

```js
it('calls provided `onChange` callback when input changes', () => {
  const onChange = sinon.stub();
  const wrapper = createComponent(Checkbox, {
    onChange,
  });
                                                                    
  const checkbox = wrapper.find('input').getDOMNode();
  const event = new Event('change');
  act(() => {
    checkbox.dispatchEvent(event);
  });
                                                                    
  assert.calledWith(onChange, event);
});
                                                                    
it('updates controlled checkbox state on "change" event', () => {
  const wrapper = createComponent(Checkbox);
  const checkbox = wrapper.find('input').getDOMNode();
                                                                    
  // Toggle the checkbox without updating the controlled state
  const toggle = () => (checkbox.checked = !checkbox.checked);
                                                                    
  // Toggle the checkbox and update the controlled state
  const simulateChange = () => {
    toggle();
    act(() => {
      checkbox.dispatchEvent(new Event('change'));
    });
  };
                                                                    
  simulateChange();
  toggle();
  wrapper.setProps({});
  assert.isTrue(checkbox.checked);
                                                                    
  simulateChange();
  toggle();
  wrapper.setProps({});
  assert.isFalse(checkbox.checked);
});
```

I split the two scenarios into two, and to test controlled state updates, relied on the fact there is a way to update the checkbox's actual state without updating `Checkbox`'s controlled state.

Reply:
Edit: I used "controlled" in the previous comment originally when I meant "uncontrolled".

Reply:
Thank you!